Khan/perseus
# FEI-4967.4: Fix or suppress all in-source type errors

## Summary:
There are still some TypeScript errors when running 'yarn typecheck'.  These are due to conflicts between globals defined by cypress and jest.  These errors will be addressed in a future PR.

I've also updated the CI workflow to run 'yarn typecheck' instead of 'yarn flow' so that we can start getting some ffeedback about TypeScript errors in the PRs.

Issue: FEI-4967

## Test plan:
- yarn typecheck
- see that all of the errors are in node_modules
- let CI run
- see that the TypeScript errors are being reported as a GitHub Actions check
